community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -545,6 +496,39 @@ public String toString()
         return "RWLock[" + resource + ", hash="+hashCode()+"]";
     }
 
+    private void acquireLock( LockType type, Object tx, TxTermination termination, TxLockElement tle )
+    {
+        ragManager.checkWaitOn( this, tx );
+        WaitElement waitElement = new WaitElement( tle, type, currentThread() );
```

Comment:
`WaitElement`s are cleaned inside `#waitForLock()` if tx was terminated. at the same time `ragManager.checkWaitOn()` modifies internal global map so not calling `ragManager.stopWaitOn()` after tx was terminated is wrong. thanks for spotting this, will fix.